drv/lpc55-update-server/src/images.rs

```diff
@@ -2,14 +2,32 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
+/// Implement LPC55_`update_server`'s knowledge about bits inside an image.
+///
+/// The update server needs to work with partial or otherwise invalid images.
+/// Signature checks are only performed at boot time. The update server
+/// does match FWIDs against the boot-time-info in some cases. But, because
+/// flash areas are mutated during update_server operations and stuff can
+/// happen, any data in the non-active Hubris image needs to be treated as
+/// untrusted (update_server does not alter its own image).
```

Comment:
**question:** Idle curiosity and paranoia, but is the active image actually write-protected as well? Or would it theoretically be possible to modify it?

Reply:
The active image could be modified by the update_server task if there were a code path to do that.
There is a check testing to see if the slot to be updated is the currently executing Hubris slot. If so, the request is rejected.